src/databricks/labs/pytester/fixtures/iam.py

```diff
@@ -94,15 +94,62 @@ def _scim_values(ids: list[str]) -> list[iam.ComplexValue]:
     return [iam.ComplexValue(value=x) for x in ids]
 
 
+def _wait_group_provisioned(interface: AccountGroupsAPI | GroupsAPI, group: Group) -> None:
+    """Wait for a group to be visible via the supplied group interface.
+
+    Due to consistency issues in the group-management APIs, new groups are not always visible in a consistent manner
+    after being created or modified. This method can be used to mitigate against this by checking that a group:
+
+     - Is visible via the `.get()` interface;
+     - Is visible via the `.list()` interface that enumerates groups.
+
+    Visibility is assumed when 2 calls in a row return the expected results.
+
+    Args:
+          interface: the group-management interface to use for checking whether the groups are visible.
+          group: the group whose visibility should be verified.
+    Raises:
+          NotFound: this is thrown if it takes longer than 90 seconds for the group to become visible via the
+          management interface.
+    """
+    # Use double-checking to try and compensate for the lack of monotonic consistency with the group-management
+    # interfaces: two subsequent calls need to succeed for us to proceed. (This is probabilistic, and not a guarantee.)
+    # The REST API internals cache things for up to 60s, and we see times close to this during tests. The retry timeout
+    # reflects this: if it's taking much longer then something else is wrong.
+
+    @retried(on=[NotFound], timeout=timedelta(seconds=90))
+    def _double_get_group(group: Group) -> None:
+        assert group.id
+        interface.get(group.id)
+        interface.get(group.id)
+
+    def _check_group_in_listing() -> None:
+        found_groups = interface.list(attributes="id", filter=f'id eq "{group.id}"')
+        found_ids = {found_group.id for found_group in found_groups}
+        if group.id not in found_ids:
+            msg = f"Group id not (yet) found in group listing: {group.id}"
+            raise NotFound(msg)
+
+    @retried(on=[NotFound], timeout=timedelta(seconds=90))
+    def _double_check_group_in_listing() -> None:
+        _check_group_in_listing()
+        _check_group_in_listing()
+
+    _double_get_group(group)
+    _double_check_group_in_listing()
+
+
 def _make_group(name: str, cfg: Config, interface, make_random, watchdog_purge_suffix) -> Generator[Group, None, None]:
+    _not_specified = object()
```

Comment:
Why not use `None` here?

Reply:
The idea here is to know that it's impossible for the caller to provide a `wait_for_provisioning=` argument. With `None` we can't guarantee that, but using a sentinel default value does guarantee it and is an approach that's sometimes used to distinguish between `None` and an unprovided value.

We care about this because I want to warn if the caller is using it irrespective of the value. (When we remove this argument then any such calls with `wait_for_provisioning` will fail with `TypeError`.)